src/compiler.ts

```diff
@@ -1483,6 +1484,12 @@ export class Compiler extends DiagnosticEmitter {
       instance.set(CommonFlags.ERRORED);
     }
 
+    instance.localsByIndex
+      .filter((local: Local) => local.envOffset >= 0)
+      .forEach((local: Local) => {
+      console.log("writing " + local.name + " into the env of " + instance.name + " at offset " + local.envOffset);
+    })
```

Comment:
This is performed post-compilation- I figured it would be simplest to gather all the closure information by compiling *then* go back and update all the binaryen statements to load from the environment instead where necessary.

Reply:
This is somewhat connected to the problem of scoped locals being shared, since if we replace too late we can encounter other accesses on the same shared local that are not actually part of the env. My intuition here is to remove sharing locals during code generation, but last time I checked this had surprising implications on other logic in that we cannot recompile blocks of code anymore without changing local order.

Reply:
When you say that the 'Local' is shared- are you referring to the compiler struct `Local` or the local index referred to in the binaryen IR?

Reply:
Currently it's both. Scoped locals can be free'd when their lifetime ends, and subsequent code can reuse one from the `tempXY` arrays,, including its index.